src/config-import.js

```diff
@@ -0,0 +1,231 @@
+import yauzl from 'yauzl-promise'
+import { validate, valueSchemas } from '@mapeo/schema'
+import { text, buffer } from 'node:stream/consumers'
+
+// Throw error if a zipfile contains more than 10,000 entries
+const MAX_ENTRIES = 10_000
+
+/**
+ * @typedef {Omit<yauzl.Entry, 'fileName'> & { filename: string }} Entry
```

Comment:
Nit: I was confused by this before I realized that `@types/yauzl-promise` is incorrect. Would you mind adding an explanation here?

One possible option:

```suggestion
 * @typedef {Omit<yauzl.Entry, 'fileName'> & { filename: string }} Entry
 * `@types/yauzl-promise` has the wrong name for this key. We should
 * eventually submit this patch upstream.
```

Feel free to ignore if you think it's obvious.

Reply:
Yeah, I don't its obvious at all, so I'll add the comment